src/dra/4A538.c

```diff
@@ -942,238 +940,247 @@ void RenderEntities(void) {
+                    if (!(animFrameFlags & 2)) {
+                        if (entity->rotZ == 0x800) {
+                            poly->x0++;
+                            poly->x1++;
+                            poly->x2++;
+                            poly->x3++;
+                        }
+                    }
+#endif
+                    func_800EB758(r->x, r->y, entity, r->eDrawFlags, poly,
+                                  r->facingLeft2);
                 }
-                temp_v1_6 = entity->palette;
-                if (temp_v1_6 & 0x8000) {
-                    var_v0_4 = temp_v1_6 & 0x7FFF;
+                palette = entity->palette;
+                if (palette & 0x8000) {
+                    poly->clut = g_ClutIds[palette & 0x7FFF];
                 } else {
-                    var_v0_4 = temp_v1_6 + var_t2[5];
+                    poly->clut = g_ClutIds[animFrame[5] + palette];
                 }
-                poly->clut = g_ClutIds[var_v0_4];
-                texSX = var_t2[7];
-                texSY = var_t2[8];
-                texEX = var_t2[9];
-                temp_v1_8 = r->unk0;
-                texEY = var_t2[10];
-                if (temp_v1_8 & 1) {
-                    texSX -= 0x80;
-                    texEX -= 0x80;
+                uvRight = animFrame[7];
+                uvTop = animFrame[8];
+                uvLeft = animFrame[9];
+                uvBottom = animFrame[10];
+                if (r->unk0 & 1) {
+                    uvRight += 0x80;
+                    uvLeft += 0x80;
                 }
-                if (temp_v1_8 & 2) {
-                    texSY -= 0x80;
-                    texEY -= 0x80;
+                if (r->unk0 & 2) {
+                    uvTop += 0x80;
+                    uvBottom += 0x80;
                 }
-                if (temp_s2_2 & 4) {
-                    texEX--;
+                if (animFrameFlags & 4) {
+                    uvLeft--;
                 }
-                if (temp_s2_2 & 8) {
-                    texEY--;
+                if (animFrameFlags & 8) {
+                    uvBottom--;
                 }
-                if (temp_s2_2 & 0x10) {
-                    texSX++;
+                if (animFrameFlags & 0x10) {
+                    uvRight++;
                 }
-                if (temp_s2_2 & 0x20) {
-                    texSY++;
+                if (animFrameFlags & 0x20) {
+                    uvTop++;
                 }
-                if (((temp_s2_2 & 2) ^ r->unk20) == 0) {
-                    if (!(temp_s2_2 & 1)) {
-                        poly->u0 = texSX;
-                        poly->v0 = texSY;
-                        poly->u1 = texEX;
-                        poly->v1 = texSY;
-                        poly->u2 = texSX;
-                        poly->v2 = texEY;
-                        poly->u3 = texEX;
-                        poly->v3 = texEY;
+                tempFlagCheck = (animFrameFlags & 2) ^ r->facingLeft2;
```

Comment:
I think you could rename `r->facingLeft2` as `r->flipYX`?

Reply:
Sure. Why `flipYX` instead of just `flipX`?

Reply:
I believe the first bit is to flip by Y? That's what I am understanding from line `1075`

Reply:
I think in `animFrameFlags`, bit 1 is to flip Y (as in 1075), and bit 2 is to flip X, but `tempFlagCheck` only samples bit 2, so `r->flipX/flipYX` only controls the X flipping?